packages/connect/src/Detector.ts

```diff
@@ -1,37 +1,40 @@
 import { ApiPromise } from '@polkadot/api';
+import { ProviderInterface } from '@polkadot/rpc-provider/types';
 import { SmoldotProvider }  from './SmoldotProvider';
 import { ExtensionProvider } from './ExtensionProvider';
 
 export class Detector {
-    #chainName: string;
-    #chainSpec: string | undefined;
-    #isExtension: boolean;
-    #provider: SmoldotProvider | ExtensionProvider | undefined;
+  #name: string;
+  #isExtension: boolean;
+  #providers: Record<string, ProviderInterface> = {};
 
-    public constructor (chainName: string, chainSpec?: string) {
-        this.#chainName = chainName;
-        this.#chainSpec = chainSpec;
-        this.#isExtension = !!document.getElementById('substrateExtension');
-    }
+  get name(): string {
+    return this.#name;
+  }
+
+  public constructor (name: string) {
+    this.#isExtension = !!document.getElementById('substrateExtension');
+    this.#name = name;
+  }
+
+  public connect = async (chainName: string, chainSpec?: string): Promise<ApiPromise> => {
```

Comment:
I thought chainspec will not be provided from Apps

Reply:
Yup but we hadn't fully decided what to do while I was working on this. Also the goal of this PR was not to deal with that .. so I just did the smallest change to satisfy what I was doing.  We can discuss and change the public API for chainspecs in a future PR

Reply:
There was a question about that also in CI channel. We should consider leaving that option - but lets discuss in later stage